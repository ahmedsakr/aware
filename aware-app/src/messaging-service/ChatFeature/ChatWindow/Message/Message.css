.received-message {
    margin-top: 10px;
    display: flex;
}

.received-message-profile {
    border: 2px solid rgb(54, 56, 58);
    border-radius: 5px;
    text-align: center;
    padding: 0px;
    margin-right: 5px;
}

.received-message-profile-img {
    max-width: 30%;
}

.received-message-profile-name {
    background-color: #bac1b8;
    text-align: center;
    margin: 0px;
}

.received-message-text {
    background-color: #a1c6ea;
    border-radius: 10px;
    padding: 5px;
    margin-right: 5px;
}

.received-message-timestamp {
    padding: 0px;
    text-align: left;
    font-size: 0.6em;
}

.sent-message {
    margin-top: 10px;
    display: flex;
    margin-right: 5px;
    padding-right: 15px;
    text-align: center;
}

.sent-message-profile {
    padding: 0px;
    font-size: 0.8em;
}

.sent-message-profile-name {
    margin: 0px;
}

.sent-message-text {
    display: flex;
    align-items: flex-end;
    text-align: left;
}

.sent-message-text > p {
    background-color: #507dbc;
    border-radius: 10px;
    padding: 5px;
    margin-left: 5px;
    width: 100%;
    margin-bottom: 0px;
}

.sent-message-timestamp {
    padding: 0px;
    text-align: right;
    font-size: 1em;
    display: flex;
    align-items: flex-end;
    justify-content: flex-end;
    opacity: 0.8;
}

.sent-message-timestamp > p {
    margin: 0;
    padding-right: 5px;
}