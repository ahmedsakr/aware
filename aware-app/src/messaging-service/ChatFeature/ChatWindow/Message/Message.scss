/*
 * Message styling
 *
 * Defines layout fundamentals for every message.
 */
 $received_message_background_color: #a1c6ea;
 $sent_message_background_color: #507dbc;


 .message {
    margin-top: 10px;
    display: flex;
  }
  
  .received-message {
    margin-left: 5px;
    padding-left: 15px;
  }
  
  .sent-message {
    margin-right: 5px;
    padding-right: 15px;
  }
  
  /*
   * Message text styling
   *
   * The messages bubbles are designed to be fluid in size relative to the size of the text.
   */
  
  .message-text {
    display: flex;
    align-items: flex-start;
    text-align: left;
  }
  
  .sent-message-text {
    justify-content: right;
  }
  
  .received-message-text {
    justify-content: left;
  }
  
  .message-text > p {
    border-radius: 10px;
    margin-bottom: 0px;
    padding: 0px 10px 0px 10px;
  }
  
  .received-message-text > p {
    background-color: $received_message_background_color;
    margin-right: 5px;
  }
  
  .sent-message-text > p {
    background-color: $sent_message_background_color;
    margin-left: 5px;
  }
  
  /*
   * Profile styling
   * 
   * Profile elements for every message include the profile picture and the first
   * name of the sender. The name is centered under the profile image.
   */
  
  .message-profile {
    padding: 0px;
    font-size: 0.8em;
    text-align: center;
  }
  
  .message-profile-name {
    margin: 0px;
  }
  
  /*
   * Timestamp styling
   * 
   * The timestamp is designed to be on the bottom level of the divider with a minor padding
   * aware from the message.
   */
  
  .message-timestamp {
    padding: 0px;
    font-size: 1em;
    display: flex;
    align-items: flex-end;
    opacity: 0.8;
  }
  
  .received-message-timestamp {
    text-align: left;
    justify-content: flex-start;
  }
  
  .sent-message-timestamp {
    text-align: right;
    justify-content: flex-end;
  }
  
  .received-message-timestamp > p {
    margin: 0;
    padding-left: 5px;
  }
  
  .sent-message-timestamp > p {
    margin: 0;
    padding-right: 5px;
  }