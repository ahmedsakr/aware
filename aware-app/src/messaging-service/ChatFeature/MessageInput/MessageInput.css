#message-input-textfield {
  font-size: 24px;
}

#message-input-insert-png, #message-input-send-img {
    height: 65px;
}

#message-input-send-img {
    vertical-align: middle;
}

#message-input {
    background-color: #eaeff1;
    text-align: left;
}

form {
    flex: 0 1 auto;
}