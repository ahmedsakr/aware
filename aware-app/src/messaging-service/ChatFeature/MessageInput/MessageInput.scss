/*
 * Message Input styling
 *
 * Defines layout fundamentals for the Message Input Field.
 */

$message_input_background_color: #eaeff1;

#message-input-textfield {
    background-color: $message_input_background_color;
    border: none;
    height: 100%;
}

#message-input-send-img {
    vertical-align: middle;
}

#message-input {
    flex: 1 0 auto;
    background-color: $message_input_background_color;
    text-align: left;
}

form {
    flex: 0 1 auto;
    display: flex;
    flex-flow: column;
}