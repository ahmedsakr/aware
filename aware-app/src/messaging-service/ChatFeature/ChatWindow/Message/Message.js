import React, { Component } from 'react';
import './Message.css'
import ProfilePicture from '../../Profile/ProfilePicture'

class Message extends Component {

    produceMessageByThem(content) {
        return (
            <div class="col-sm-12 message received-message">
                <div class="col-sm-1 message-profile">
                    <ProfilePicture instance="message" activity="online"  picture={content.avatar}/>
                    <p class="message-profile-name">{content.studentName}</p>
                </div>

                <div class="col-sm-4 message-text received-message-text">
                    <p>{content.text}</p>
                </div>

                <div class="col-sm-2 message-timestamp received-message-timestamp">
                    <p>{content.timestamp}</p>
                </div>
            </div>
        );
    }

    produceMessageByMe(content) {
        return (
            <div class="col-sm-12 message sent-message">
                <div class="offset-sm-5 col-sm-2 message-timestamp sent-message-timestamp">
                    <p>{content.timestamp}</p> 
                </div>
                <div class="col-sm-4 message-text sent-message-text">
                    <p>{content.text}</p>
                </div>
                <div class="col-sm-1 message-profile">
                    <ProfilePicture instance="message" activity="online"  picture={content.avatar}/>
                    <p class="message-profile-name">{content.studentName}</p>
                </div>
            </div>
        );
    }
    
    render() {
        if (this.props.content.studentName == this.props.name) {
            return this.produceMessageByMe(this.props.content);
        } else {
            return this.produceMessageByThem(this.props.content);
        }
    }
}

export default Message;
