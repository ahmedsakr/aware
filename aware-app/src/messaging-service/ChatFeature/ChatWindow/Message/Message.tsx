import React from 'react';
import './Message.css'
import ProfilePicture from '../../Profile/ProfilePicture'

export interface UserMessageContents {
    username: string,
    avatar: string,
    content: string,
    timestamp: string,
};

type MessageProps = {
    content: UserMessageContents,
    name: string
};

type MessageState = { };

export default class Message extends React.Component<MessageProps, MessageState> {

    produceMessageByThem(content: UserMessageContents): JSX.Element {
        return (
            <div className="col-sm-12 message received-message">
                <div className="col-sm-1 message-profile">
                    <ProfilePicture instance="message" activity="online"  picture={content.avatar}/>
                    <p className="message-profile-name">{content.username}</p>
                </div>

                <div className="col-sm-4 message-text received-message-text">
                    <p>{content.content}</p>
                </div>

                <div className="col-sm-2 message-timestamp received-message-timestamp">
                    <p>{content.timestamp}</p>
                </div>
            </div>
        );
    }

    produceMessageByMe(content: UserMessageContents): JSX.Element {
        return (
            <div className="col-sm-12 message sent-message">
                <div className="offset-sm-5 col-sm-2 message-timestamp sent-message-timestamp">
                    <p>{content.timestamp}</p> 
                </div>
                <div className="col-sm-4 message-text sent-message-text">
                    <p>{content.content}</p>
                </div>
                <div className="col-sm-1 message-profile">
                    <ProfilePicture instance="message" activity="online"  picture={content.avatar}/>
                    <p className="message-profile-name">{content.username}</p>
                </div>
            </div>
        );
    }
    
    render() {
        if (this.props.content.username === this.props.name) {
            return this.produceMessageByMe(this.props.content);
        } else {
            return this.produceMessageByThem(this.props.content);
        }
    }
}
