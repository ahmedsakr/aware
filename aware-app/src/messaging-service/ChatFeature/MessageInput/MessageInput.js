import React, {Component} from 'react';
import './MessageInput.css';
import {produceTimestamp} from '../../../aware-utils.js'

class MessageInput extends Component {
  constructor() {
    super()
    this.state = {message: ''}

    //Need to bind 'this' to give access to keyword
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  render() {
    return (
      <form onSubmit={this.handleSubmit}>
        <div id="message-input">
          <input class="col-11" id="message-input-textfield" type="text" name="message" placeholder="Type here.." onChange={this.handleChange} value={this.state.message} />
          <input class="col-1" id="message-input-send-img" type="image" src={process.env.PUBLIC_URL + "/icons8-send-letter-80.png"} alt="SEND" disabled={!this.state.message} />
        </div>
      </form>
    );
  }

  handleChange(e) {
    this.setState({
      //Set state variable to input value
      message: e.target.value
    })
  }

  handleSubmit(e) {
    //Prevents page refresh when form submit
    e.preventDefault();
    // Build a message object to be sent
    const messageObj = {
      studentName: this.props.name,
      text: this.state.message,
      timestamp: produceTimestamp(),
      avatar: "/" + this.props.name + "-pic.jpg"
    };
    this.props.sendMessage(messageObj);

    //Reset state of input to erase previous message
    this.setState({
      message: ''
    })
  }
}

export default MessageInput;
