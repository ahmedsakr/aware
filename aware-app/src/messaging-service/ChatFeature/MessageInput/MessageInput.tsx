import React from 'react';
import './MessageInput.css';
import {UserMessageContents} from '../ChatWindow/Message/Message'
import produceTimestamp from '../../../aware-utils'

type MessageInputProps = {
  sendMessage: (message: UserMessageContents) => void,
  name: string
};

type MessageInputState = {
  message: string
};

export default class MessageInput extends React.Component<MessageInputProps, MessageInputState> {
  constructor(props: MessageInputProps) {
    super(props)

    this.state = {
      message: ''
    };

    // Need to bind 'this' to give access to keyword
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleChange(event: React.ChangeEvent<HTMLInputElement>): void {
    this.setState({
      //Set state variable to input value
      message: event.target.value
    })
  }

  handleSubmit(event: React.FormEvent<HTMLFormElement>): void {
    //Prevents page refresh when form submit
    event.preventDefault();

    // Build a message object to be sent
    let messageObj: UserMessageContents = {
      username: this.props.name,
      content: this.state.message,
      timestamp: produceTimestamp(),
      avatar: "/" + this.props.name + "-pic.jpg"
    };

    this.props.sendMessage(messageObj);

    //Reset state of input to erase previous message
    this.setState({
      message: ''
    })
  }

  render(): JSX.Element {
    return (
      <form onSubmit={this.handleSubmit}>
        <div id="message-input">
          <input className="col-11" id="message-input-textfield" type="text" name="message" placeholder="Type here.." onChange={this.handleChange} value={this.state.message} />
          <input className="col-1" id="message-input-send-img" type="image" src={process.env.PUBLIC_URL + "/icons8-send-letter-80.png"} alt="SEND" disabled={!this.state.message} />
        </div>
      </form>
    );
  }
}
