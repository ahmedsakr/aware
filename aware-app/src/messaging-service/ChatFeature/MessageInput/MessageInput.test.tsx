import React from 'react';
import { shallow } from 'enzyme';

import MessageInput from './MessageInput';
import { UserMessageContents } from '../ChatWindow/Message/Message';

describe('MessageInput', () => {
    it('should render correctly', () => {

        // Stub out the function for now, but we probably need to
        // implement better tests in the future.
        let sendMessage = (message: UserMessageContents) => { };

        const component = shallow(
            <MessageInput
                sendMessage={sendMessage}
                name="myusername" />
        );

        expect(component).toMatchSnapshot();
    });
});