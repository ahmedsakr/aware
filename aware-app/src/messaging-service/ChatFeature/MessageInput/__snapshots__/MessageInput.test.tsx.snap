version https://git-lfs.github.com/spec/v1
oid sha256:4ecac46459304045259593fe66f92ec62fbdb064b240aaea0504a9e8bb8a739b
size 549
